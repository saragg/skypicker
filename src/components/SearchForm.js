import React from 'react';
import PropTypes from 'prop-types';
import RaisedButton from 'material-ui/RaisedButton';
import AutoComplete from 'material-ui/AutoComplete';
import DatePicker from 'material-ui/DatePicker';
import FontIcon from 'material-ui/FontIcon';
import MenuItem from 'material-ui/MenuItem';
import Api from '../services/Api';
import moment from 'moment';
import './SearchForm.css';

const DateTimeFormat = global.Intl.DateTimeFormat;

let timeout = null;
class SearchForm extends React.Component{
  

  constructor(props) {
    super(props);

    this.state = {
      flyFrom: null,
      to: null,
      dateFrom: null,
      dateTo: null,
      returnFrom: null,
      returnTo: null,
      optionsFrom: [],
      optionsTo: [],
      typeFlight: "round",
      errors: {},
    };

    this.handleDateFromChange = this.handleDateFromChange.bind(this);
    this.handleDateToChange = this.handleDateToChange.bind(this);
    this.handleFlyFromChange = this.handleFlyFromChange.bind(this);
    this.handleToChange = this.handleToChange.bind(this);
    this.validateForm = this.validateForm.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.searchFromPlaces = this.searchFromPlaces.bind(this);
    this.searchToPlaces = this.searchToPlaces.bind(this);
    this.searchPlaces = this.searchPlaces.bind(this);
    this.getItemComponent = this.getItemComponent.bind(this);
    this.disableFromDates = this.disableFromDates.bind(this);
    this.disableToDates = this.disableToDates.bind(this);
  }

  handleDateFromChange(event, date) {
    this.setState({
      dateFrom: moment(date).format('DD/MM/YYYY'),
      dateTo: moment(date).format('DD/MM/YYYY'),
      errors: {}
    });
  }

  handleDateToChange(event, date) {
    this.setState({
      returnFrom: moment(date).format('DD/MM/YYYY'),
      returnTo: moment(date).format('DD/MM/YYYY'),
      errors: {}
    });
  }

  handleFlyFromChange(chosenRequest, index) {
    this.setState({
      flyFrom: chosenRequest.id,
      errors: {}
    });
  }

  handleToChange(chosenRequest, index) {
    this.setState({
      to: chosenRequest.id,
      errors: {}
    });
  }

  validateForm(fields) {
    const errors = {}

    Object.keys(fields).map((field) => {
      if(!fields[field]){
        if(field === "flyFrom" || field === "to") {
          errors[field] = 'Get an option from the list'
        }
        else {
          errors[field] = 'This field is required'
        }
      }
    })

    this.setState({errors})

    return errors
  }

  handleSubmit(event) {
    event.preventDefault()

    const {flyFrom, to, dateFrom, dateTo, returnFrom, returnTo, typeFlight} = this.state

    const errors = this.validateForm({flyFrom, to, dateFrom, dateTo})
    console.log(errors)
    if(Object.keys(errors).length === 0) {
      this.props.searchFlights({flyFrom, to, dateFrom, dateTo, returnFrom, returnTo, typeFlight})
    }
  }

  searchFromPlaces(searchText, dataSource, params){
    this.searchPlaces(searchText, "optionsFrom")
  }

  searchToPlaces(searchText, dataSource, params) {
    this.searchPlaces(searchText, "optionsTo")
  }

  getItemComponent(opt) {
    let Icon;
    let primaryText;
    let countryCode;
    if(opt.type === 0) { // Airport
      Icon = <FontIcon className="fa fa-plane" />;
      primaryText = `${opt.value}(${opt.id})`;
      countryCode = opt.parentId.split("_").pop().toUpperCase();
    }
    else if(opt.type === 1) { //Country
      Icon = <FontIcon className="fa fa-map-marker" />;
      primaryText = opt.value;
      countryCode = opt.parentId;
    }
    else if(opt.type === 2) { //City
      Icon = <FontIcon className="fa fa-building" />;
      primaryText = opt.value;
      countryCode = opt.parentId;
    }
    return (
      <MenuItem
        primaryText={primaryText}
        secondaryText={countryCode}
        leftIcon={Icon}
      />
    )
  }

  searchPlaces(searchText, key){
    if (searchText.length >= 1) {
      if (timeout) {  
        clearTimeout(timeout);
      }
      timeout = setTimeout(() => {
        Api.getPlaces({term: searchText})
        .then((response)=> {

          const items = response.map((opt) => {
              const ItemComponent = this.getItemComponent(opt)
              return {
                id: opt.id,
                text: opt.value,
                value: ItemComponent,
              }
          })

          this.setState({
            [key]: items,
          })
        })
      }, 500)
    }
    else {
      this.setState({
        [key]: [],
      })
    }
  }

  disableFromDates(date){
    return moment(date).isBefore(moment().subtract(1, "day"));
  }
  disableToDates(date){
    return moment(date).isBefore(moment(this.state.dateFrom).subtract(1, "day"));
  }


  render () {

    const {optionsFrom, optionsTo, errors} = this.state

    const icons = [{name: 'hola'},{name: 'que'}, {name: 'tal'}]

    const listOfIcon = icons.map((icon) => {
        return {
          text: icon.name,
          value: <MenuItem primaryText={icon.name} secondaryText={icon.name} leftIcon={<FontIcon className="fa fa-plane" />} />,
          
        }
    });

    console.log(listOfIcon, optionsFrom)

    return (

      <form onSubmit={this.handleSubmit}>
        <div className="search-form">
          <div className="half">
            <AutoComplete
              name="flyFrom"
              floatingLabelText="Origin"
              onUpdateInput={this.searchFromPlaces}
              onNewRequest={this.handleFlyFromChange}
              dataSource={optionsFrom}
              errorText={errors["flyFrom"]}
              openOnFocus={true}
              menuProps={{autoWidth: true}}
              filter={AutoComplete.fuzzyFilter}
            />
            <AutoComplete
              name="to"
              floatingLabelText="Destination"
              onUpdateInput={this.searchToPlaces}
              onNewRequest={this.handleToChange}
              dataSource={optionsTo}
              errorText={errors["to"]}
              openOnFocus={true}
              menuProps={{autoWidth: true}}
              filter={AutoComplete.fuzzyFilter}
            />
          </div>
          <div className="half">
            <DatePicker
              name="dateFrom"
              floatingLabelText="Departure date"
              container="inline"
              mode="landscape"
              autoOk={true}
              onChange={this.handleDateFromChange}
              formatDate={new DateTimeFormat('en-US', {
                day: 'numeric',
                month: 'long',
                year: 'numeric',
              }).format}
              errorText={errors["dateFrom"]}
              shouldDisableDate={this.disableFromDates}
            />
            <DatePicker
              name="dateTo"
              floatingLabelText="Return date"
              container="inline"
              mode="landscape"
              autoOk={true}
              onChange={this.handleDateToChange}
              formatDate={new DateTimeFormat('en-US', {
                day: 'numeric',
                month: 'long',
                year: 'numeric',
              }).format}
              errorText={errors["dateTo"]}
              shouldDisableDate={this.disableToDates}
            />
          </div>
          
        </div>
        <div style={{"textAlign": "center"}}>
          <RaisedButton
            icon={<FontIcon className="fa fa-search" />}
            backgroundColor="#a4c639"
            label="Search"
            type="Submit"
            secondary={true}
          />
        </div>


      </form>
    )
  }
}

SearchForm.propTypes = {
  searchFlights: PropTypes.func.isRequired,
};

export default SearchForm;