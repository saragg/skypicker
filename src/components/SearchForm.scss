form {
    width: 100%;
    background: rgba(255,255,255,0.6);
    padding: 20px;
}
.search-form {
  margin-bottom: 20px;
  width: 100%;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  .half {
    width: 100%;
    display: -webkit-flex;
    display: flex;
    -webkit-justify-content: space-around;
    justify-content: space-around;
    @media (max-width: 624px) {
        flex-direction: column;
        text-align: -webkit-center;
    }
  }
  @media (max-width: 1080px) {
    flex-direction: column;
  }
}
