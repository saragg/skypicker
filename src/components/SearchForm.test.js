import React from 'react';
import SearchForm from './SearchForm';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

test('Renders correctly', () => {

  const searchFlights = () => {
    return;
  }

  const tree = renderer.create(
    <MuiThemeProvider>
      <SearchForm searchFlights={searchFlights}/>
    </MuiThemeProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});