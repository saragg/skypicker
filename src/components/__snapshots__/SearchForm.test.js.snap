// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="search-form"
  >
    <div
      className="half"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "position": "relative",
            "width": 256,
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <label
            className={undefined}
            htmlFor="flyFrom-undefined-Origin-3186"
            onTouchTap={undefined}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Origin
          </label>
          <input
            autoComplete="off"
            disabled={false}
            id="flyFrom-undefined-Origin-3186"
            name="flyFrom"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            open={false}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "position": "relative",
            "width": 256,
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <label
            className={undefined}
            htmlFor="to-undefined-Destination-38397"
            onTouchTap={undefined}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Destination
          </label>
          <input
            autoComplete="off"
            disabled={false}
            id="to-undefined-Destination-38397"
            name="to"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            open={false}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
    <div
      className="half"
    >
      <div
        className={undefined}
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <label
            className={undefined}
            htmlFor="dateFrom-undefined-Departuredate-40479"
            onTouchTap={undefined}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Departure date
          </label>
          <input
            disabled={false}
            id="dateFrom-undefined-Departuredate-40479"
            name="dateFrom"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          />
        </div>
      </div>
      <div
        className={undefined}
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <label
            className={undefined}
            htmlFor="dateTo-undefined-Returndate-14689"
            onTouchTap={undefined}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Return date
          </label>
          <input
            disabled={false}
            id="dateTo-undefined-Returndate-14689"
            name="dateTo"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#ff4081",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="Submit"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              className="fa fa-search"
              color="#ffffff"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fontSize": 24,
                  "marginLeft": 12,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                }
              }
            />
            <span
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 8,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Search
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</form>
`;
